src/plugins/data/public/ui/filter_bar/filter_item.tsx

```diff
@@ -154,83 +157,90 @@ export function FilterItem(props: Props) {
 
   function getPanels() {
     const { negate, disabled } = filter.meta;
-    return [
+    let mainPanelItems = [
```

Comment:
Can we type this `EuiContextMenuPanelItemDescriptor & {id: PanelOptions}`  and then use `id` for filtering `hiddenPanelOptions` 🙏  

Would be nice to have `hiddenPanelOptions` type connected to actually panel items and not use an arbitrary `data-test-subj` string for this

Reply:
i tried adding id or using EuiContextMenuPanelItemDescriptor, typescript goes crazy in both cases.

Reply:
working with EUI items types can be tricky , so i am not sure, how to solve this.

Reply:
Wow, indeed tricky!

To workaround eui types, I ended up with: 


```
type PanelOptions = 'pinFilter' | 'editFilter' | 'negateFilter' | 'disableFilter' | 'deleteFilter';

interface PanelItem {
  id: PanelOptions;
  euiPanel: EuiContextMenuPanelItemDescriptor;
}

....

let mainPanelItems: PanelItem[] = [
      {
        id: 'pinFilter',
        euiPanel: {
          name: isFilterPinned(filter)
            ? props.intl.formatMessage({
                id: 'data.filter.filterBar.unpinFilterButtonLabel',
                defaultMessage: 'Unpin',
              })
            : props.intl.formatMessage({
                id: 'data.filter.filterBar.pinFilterButtonLabel',
                defaultMessage: 'Pin across all apps',
              }),
          icon: 'pin',
          onClick: () => {
            setIsPopoverOpen(false);
            onTogglePinned();
          },
          'data-test-subj': 'pinFilter',
        },
      },
....
    ];
```


And then filter panels by `id: 'pinFilter',`


to be honest, I still think it is worth it

Reply:
^^ i don't think that will work.

Reply:
so i am leaving this improvement perhaps for future :)